uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/zones/ZoneControllerMockMvcTests.java

```diff
@@ -0,0 +1,249 @@
+            assertNotNull(zoneResponse);
+            assertNotNull(identityZone);
+            assertNotNull(zoneResponse.getParameters());
+            assertEquals(identityZone.getSubdomain(), zoneResponse.getParameters().getSubdomain());
+            assertEquals(identityZone.getSubdomain(), zoneResponse.getConnectionDetails().getSubdomain());
+            String uri = "http://" + identityZone.getSubdomain() + ".localhost";
+            assertEquals(uri, zoneResponse.getConnectionDetails().getUri());
+            assertEquals("http://localhost:8080/dashboard", zoneResponse.getConnectionDetails().getDashboardUri());
+            assertEquals(uri + "/oauth/token", zoneResponse.getConnectionDetails().getIssuerId());
+            assertEquals(X_IDENTITY_ZONE_ID, zoneResponse.getConnectionDetails().getZone().getHttpHeaderName());
+            assertEquals(identityZone.getId(), zoneResponse.getConnectionDetails().getZone().getHttpHeaderValue());
+        }
+    }
+
+    @Test
+    void testGetZone_Notfound() throws Exception {
+        MvcResult result = mockMvc.perform(
+                   get("/zones").param("name", "random-name")
+                                .header("Authorization", "Bearer " + identityClientZonesReadToken))
+               .andExpect(status().isNotFound()).andReturn();
+        assertEquals(
+            "{\"message\", \"Zone[random-name] not found.\" }",
+            result.getResponse().getContentAsString());
+    }
+
+    //TODO: delete once the orchestrator create API implemented
+    @SneakyThrows
+    private IdentityZone createIdentityZone() {
+        IdentityZone identityZone = createSimpleIdentityZone(RandomString.make(10));
+        IdentityZoneConfiguration identityZoneConfiguration = new IdentityZoneConfiguration();
+        Map<String, String> keys = new HashMap<>();
+        keys.put("kid", "key");
+        identityZoneConfiguration.getTokenPolicy().setKeys(keys);
+        identityZoneConfiguration.getTokenPolicy().setActiveKeyId("kid");
+        identityZoneConfiguration.getTokenPolicy().setKeys(keys);
+
+        identityZone.setConfig(identityZoneConfiguration);
+        identityZone.getConfig().getSamlConfig().setPrivateKey(serviceProviderKey);
+        identityZone.getConfig().getSamlConfig().setPrivateKeyPassword(serviceProviderKeyPassword);
+        identityZone.getConfig().getSamlConfig().setCertificate(serviceProviderCertificate);
+        MvcResult result = mockMvc.perform(
+                                      post("/identity-zones")
+                                          .header("Authorization", "Bearer " + uaaAdminClientToken)
+                                          .contentType(APPLICATION_JSON)
+                                          .content(JsonUtils.writeValueAsString(identityZone)))
+                                  .andExpect(status().is(HttpStatus.CREATED.value()))
+                                  .andReturn();
+
+        if (HttpStatus.OK.is2xxSuccessful()) {
```

Comment:
Same here, Why if condition? if it is creating zone, then i guess we can remove if check

Reply:
Removed.